packages/engine/src/datastore/table/proxy.rs

```diff
@@ -83,34 +85,36 @@ pub struct BatchWriteProxy<K: Batch> {
 
 impl<K: Batch> BatchWriteProxy<K> {
     pub fn new(arc: &Arc<RwLock<K>>) -> Result<BatchWriteProxy<K>> {
-        if unsafe { arc.raw() }.try_lock_exclusive() {
+        // Safety: `try_lock_exclusive` locks the `RawRwLock` and returns `bool` instead of a guard.
+        if unsafe { RwLock::raw(arc).try_lock_exclusive() } {
             Ok(BatchWriteProxy { arc: arc.clone() })
         } else {
             Err(Error::from("Couldn't acquire exclusive lock on object"))
         }
     }
 
-    pub fn inner_mut(&mut self) -> &mut K {
+    pub fn batch(&self) -> &K {
         let ptr = self.arc.data_ptr();
         // Safety: This method uses unsafe code to get a mutable reference to the underlying data.
-        // However since this object acts as a guarantee of no other read/write locks existing,
-        // then data races stemming from the dereferencing happening within this method cannot
-        // happen
-        unsafe { &mut *ptr }
+        // However since this object acts as a guarantee of no other read/write locks existing, then
+        // data races stemming from the dereferencing happening within this method cannot happen
+        unsafe { &*ptr }
     }
 
-    pub fn inner(&self) -> &K {
+    pub fn batch_mut(&mut self) -> &mut K {
         let ptr = self.arc.data_ptr();
         // Safety: This method uses unsafe code to get a mutable reference to the underlying data.
-        // However since this object acts as a guarantee of no other read/write locks existing, then
-        // data races stemming from the dereferencing happening within this method cannot happen
-        unsafe { &*ptr }
+        // However since this object acts as a guarantee of no other read/write locks existing,
+        // then data races stemming from the dereferencing happening within this method cannot
+        // happen
+        unsafe { &mut *ptr }
     }
 }
 
 impl<K: Batch> Drop for BatchWriteProxy<K> {
+    // SAFETY: We are holding a shared lock, so we can unlock it
```

Comment:
Is this not holding an exclusive lock at this point?

Reply:
Yes, Copy-Paste error, thanks. I also changed the description to be the same as in `BatchReadProxy::drop`.